src/libraries/System.Linq.Expressions/tests/BinaryOperators/Assignment/OpAssign.cs

```diff
@@ -605,5 +605,36 @@ public void ConvertOpWriteByRefParameterOverloadedOperator(bool useInterpreter)
             AddsToSomethingElse result = del(ref arg, new AddsToSomethingElse(35));
             Assert.Equal(result, arg);
         }
+
+        private delegate void RightShiftAssignDelegate<T>(ref T left, int right);
+
+        [Fact]
+        public void RightShiftAssign()
+        {
+            rightShiftAssign<sbyte>(unchecked((sbyte)0x80), 7, (sbyte)-1);
+            rightShiftAssign<byte>(unchecked((byte)0x80), 7, (byte)1);
+            rightShiftAssign<short>(unchecked((short)0x8000), 15, (short)-1);
+            rightShiftAssign<ushort>(unchecked((ushort)0x8000), 15, (ushort)1);
+            rightShiftAssign<int>(unchecked((int)0x8000_0000), 31, (int)-1);
+            rightShiftAssign<uint>(unchecked((uint)0x8000_0000), 31, (uint)1);
+            rightShiftAssign<long>(unchecked((long)0x8000_0000_0000_0000), 63, (long)-1);
+            rightShiftAssign<ulong>(unchecked((ulong)0x8000_0000_0000_0000), 63, (ulong)1);
+
+            static void rightShiftAssign<T>(T left, int right, T expected)
+            {
+                var leftOperand = Expression.Parameter(typeof(T).MakeByRefType());
+                var rightOperand = Expression.Parameter(typeof(int));
+                var expr = Expression.RightShiftAssign(leftOperand, rightOperand);
+                var lambda = Expression.Lambda<RightShiftAssignDelegate<T>>(expr, leftOperand, rightOperand);
+                compileAndInvoke(lambda, left, right, expected, preferInterpretation: false);
+                compileAndInvoke(lambda, left, right, expected, preferInterpretation: true);
```

Comment:
Did u consider making this test a `[Theory]` and using `preferInterpretation` as the test parameter?

Reply:
> Did u consider making this test a [Theory] and using preferInterpretation as the test parameter?

Yes. I decided to include the cases explicitly since the test already includes explicit cases for the various primitive types.